modules/ximgproc/src/rolling_guidance_filter.cpp

```diff
@@ -59,18 +57,24 @@ namespace ximgproc
 
         dst_.create(src.size(), src.type());
         Mat dst = dst_.getMat();
+        Mat& guidance = dst;  // dst and guidance are the same
 
-        if (src.data == guidance.data)
-            guidance = guidance.clone();
         if (dst.data == src.data)
             src = src.clone();
 
         int srcCnNum = src.channels();
 
         if (srcCnNum == 1 || srcCnNum == 3)
         {
-            while(numOfIter--){
-                jointBilateralFilter(guidance, src, guidance, d, sigmaColor, sigmaSpace, borderType);
+            if (numOfIter > 0) {
```

Comment:
gaussianBlur must be used only to initialize image

Reply:
Yes, gaussianBlur is used only at iteration 0.
This is intended according to Algorithm 1 in p. 10 of [the paper](http://www.cse.cuhk.edu.hk/leojia/projects/rollguidance/paper/[ECCV2014]RollingGuidanceFilter_5M.pdf).
The filter returns a constant image when `numOfIter==0`, and returns a gaussianBlur result when `numOfIter==1`.